tests/helpers/test_fsutil.py

```diff
@@ -210,6 +221,369 @@ def test_stat_result():
     assert st == my_stat
 
 
+@pytest.fixture
+def path_fs() -> Iterator[VirtualFilesystem]:
+    vfs = VirtualFilesystem()
+
+    vfs.makedirs("/some/dir")
+    vfs.symlink("/some/dir/file.txt", "/some/symlink.txt")
+    vfs.symlink("nonexistent", "/some/dir/link.txt")
+    vfs.map_file_fh("/some/file.txt", io.BytesIO(b"content"))
+    vfs.map_file_fh("/some/dir/file.txt", io.BytesIO(b""))
+
+    yield vfs
+
+
+def test_target_path_drive(path_fs: VirtualFilesystem) -> None:
+    assert path_fs.path("/some/file.txt").drive == ""
+
+
+def test_target_path_root(path_fs: VirtualFilesystem) -> None:
+    assert path_fs.path("/some/file.txt").root == "/"
+
+
+def test_target_path_anchor(path_fs: VirtualFilesystem) -> None:
+    assert path_fs.path("/some/file.txt").anchor == "/"
+
+
+def test_target_path_parent(path_fs: VirtualFilesystem) -> None:
+    assert path_fs.path("/some/dir/file.txt").parent == path_fs.path("/some/dir")
+
+
+def test_target_path_parents(path_fs: VirtualFilesystem) -> None:
+    path = path_fs.path("/some/dir/file.txt")
+    parents = list(path.parents)
+    assert parents == [path_fs.path("/some/dir"), path_fs.path("/some"), path_fs.path("/")]
+    assert [p.exists() for p in parents]
+    assert all([p._fs == path_fs for p in parents])
+
+
+def test_target_path_name(path_fs: VirtualFilesystem) -> None:
+    path = path_fs.path("/some/file.txt")
+    assert path.name == "file.txt"
+
+
+def test_target_path_suffix(path_fs: VirtualFilesystem) -> None:
+    path = path_fs.path("/some/file.txt")
+    assert path.suffix == ".txt"
+
+
+def test_target_path_suffixes(path_fs: VirtualFilesystem) -> None:
+    path = path_fs.path("/some/file.tar.gz")
+    assert path.suffixes == [".tar", ".gz"]
+
+
+def test_target_path_stem(path_fs: VirtualFilesystem) -> None:
+    path = path_fs.path("/some/file.txt")
+    assert path.stem == "file"
+
+
+def test_target_path_as_posix(path_fs: VirtualFilesystem) -> None:
+    # TODO windows
+    path = path_fs.path("/some/file.txt")
+    assert path.as_posix() == "/some/file.txt"
+
+
+def test_target_path_as_uri(path_fs: VirtualFilesystem) -> None:
+    path = path_fs.path("/some/file.txt")
+    assert path.as_uri() == "file:///some/file.txt"
+
+
+def test_target_path_is_absolute(path_fs: VirtualFilesystem) -> None:
+    assert path_fs.path("/some/file.txt").is_absolute()
+    assert not path_fs.path("some/file.txt").is_absolute()
+
+
+def test_target_path_is_relative_to(path_fs: VirtualFilesystem) -> None:
+    assert path_fs.path("/some/dir/file.txt").is_relative_to("/some/dir")
+    assert not path_fs.path("/some/dir/file.txt").is_relative_to("/some/other")
+
+
+def test_target_path_is_reserved(path_fs: VirtualFilesystem) -> None:
+    # TODO
+    assert not path_fs.path("CON").is_reserved()
```

Comment:
Maybe make the test as it should be, but add a skip decorator instead?

Reply:
Forgot about this TODO, added a comment instead. We do not have any reserved names atm, so this should just always return `False`.